searx/engines/dailymotion.py

```diff
@@ -54,45 +106,61 @@ def response(resp):
 
```

Comment:
There is no list when there is an error: we can log this error (see qwant.py), so debugging is going to be easier.

Reply:
From cdecb87478bb114b4f74f418ed4afb244e32b138 :

```python
def request(query, params):
    ....
    params['raise_for_httperror'] = False

    return params

def response(resp):
    results = []

    search_res = resp.json()

    # check for an API error
    if 'error' in search_res:
        raise SearxEngineAPIException(search_res['error'].get('message'))

    # raise for other errors
    raise_for_httperror(resp)
```

How to test: query for `!dailymotion`, since the query sent to dailymotion is empty, the API returns an error (the bug should not be fix in this PR).

Do note the usage of `search_res['error'].get('message')` which provides a meaningful error message.

Reply:
Oops I have overseen your additional commits ..

> Do note the usage of `search_res['error'].get('message')` which provides a meaningful error message.

I squashed your https://github.com/searxng/searxng/commit/cdecb87478bb114b4f74f418ed4afb244e32b138 into my commit 3bb62823 and modified the request function ... 

> How to test: query for !dailymotion, since the query sent to dailymotion is empty,

Instead of sending an unsupported request (empty query string) to daylimotion, additionally I return `False` from the request function if query string is empty ..  https://github.com/searxng/searxng/pull/1071/files#diff-f3739fa1a440898cb603ad679a81c9429d1e74409852c45dd2c8e5f595c98aa0R81-R82

---

If you think PR is now ready .. go ahead and merge ..